### PR TITLE
Update template to remove clippy warnings

### PR DESCRIPTION
**Pull request by simonsso**
Using this template, generated weight files contains clippy warnings like this:

```
    |
133 |             .saturating_add(T::DbWeight::get().writes((1 as u64).saturating_mul(r as u64)))
    |                                                       ^^^^^^^^^^ help: try: `1_u64`
    |
```

Updating template and recreating benchmarks will remove the warning. 
Hope this does not break anything elsewhere.

_Edit:_
This problem was detected and solved in another project using this template as a template for benchmarks.

### CODE REVIEW

File: utils/frame/benchmarking-cli/src/pallet/template.hbs [hunk 1]
```diff
@@ -34,22 +34,22 @@ impl<T: frame_system::Config> {{pallet}}::WeightInfo for WeightInfo<T> {
 		{{~#if (not c.is_used)}}_{{/if}}{{c.name}}: u32, {{/each~}}
 	) -> Weight {
 		// Minimum execution time: {{underscore benchmark.min_execution_time}} nanoseconds.
-		Weight::from_ref_time({{underscore benchmark.base_weight}} as u64)
+		Weight::from_ref_time({{underscore benchmark.base_weight}}_u64)
```
Comment (bkchr) [inline utils/frame/benchmarking-cli/src/pallet/template.hbs:37]:
```suggestion
		Weight::from_ref_time({{underscore benchmark.base_weight}})
```

The compiler should be able to defer this automatically here and for the other use cases of raw numbers.

**Comment by ggwpez** [comment]
/cmd queue -c bench-bot $ pallet dev pallet_lottery

**Comment by simonsso** [comment]
> @ggwpez Command `"$PIPELINE_SCRIPTS_DIR/bench-bot.sh" pallet dev pallet_lottery` has finished. Result: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2033725 has finished. If any artifacts were generated, you can download them from https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2033725/artifacts/download.

This command uses template file: `./.maintain/frame-weight-template.hbs`
The patch changes the template template `utils/frame/benchmarking-cli/src/pallet/template.hbs`

File: .maintain/frame-weight-template.hbs [hunk 2]
```diff
@@ -49,22 +49,22 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
...
-			.saturating_add(T::DbWeight::get().reads(({{cr.slope}} as u64).saturating_mul({{cr.name}} as u64)))
+			.saturating_add(T::DbWeight::get().reads(({{cr.slope}}).saturating_mul({{cr.name}} as u64)))
 			{{/each}}
 			{{#if (ne benchmark.base_writes "0")}}
-			.saturating_add(T::DbWeight::get().writes({{benchmark.base_writes}} as u64))
+			.saturating_add(T::DbWeight::get().writes({{benchmark.base_writes}}))
 			{{/if}}
 			{{#each benchmark.component_writes as |cw|}}
-			.saturating_add(T::DbWeight::get().writes(({{cw.slope}} as u64).saturating_mul({{cw.name}} as u64)))
+			.saturating_add(T::DbWeight::get().writes(({{cw.slope}}_u64).saturating_mul({{cw.name}} as u64)))
```
Comment (bkchr) [inline .maintain/frame-weight-template.hbs:67]:
```suggestion
			.saturating_add(T::DbWeight::get().writes(({{cw.slope}}_u64).saturating_mul({{cw.name}})))
```

Here we also don't need this.

**Review by bkchr — APPROVED** [review]

File: .maintain/frame-weight-template.hbs (hunk 2 above)
Comment (simonsso) [reply .maintain/frame-weight-template.hbs:67]: @bkchr  Don't you mean:
```
.saturating_add(T::DbWeight::get().writes(({{cw.slope}}).saturating_mul({{cw.name}} as u64)))
```
To be aligned with the other lines?

**Comment by simonsso** [comment]
@ggwpez 
> /cmd queue -c bench-bot $ pallet dev pallet_lottery

I think this could be rerun now with the changes to the other template file

File: .maintain/frame-weight-template.hbs (hunk 2 above)
Comment (bkchr) [reply .maintain/frame-weight-template.hbs:67]: Ahh yeah :facepalm: Updated the suggested change.

**Comment by simonsso** [comment]
/cmd queue -c bench-bot $ pallet dev pallet_lottery

**Comment by simonsso** [comment]
> @simonsso Requester could not be detected as a member of an allowed organization.
Aha that is how it works ;-)

> /cmd queue -c bench-bot $ pallet dev pallet_lottery

@ggwpez  Could you please rerun above to get a valid output with the updated template.

**Comment by ggwpez** [comment]
There are still some `RocksDbWeight::get().reads(({{cr.slope}} as u64).` in the lower half of `.maintain/frame-weight-template.hbs` since it contains the generation macro code twice.  
Could you try your linter on the output that the weight bot pushed please?

**Comment by simonsso** [comment]
> There are still some `RocksDbWeight::get().reads(({{cr.slope}} as u64).` in the lower half of `.maintain/frame-weight-template.hbs` since it contains the generation macro code twice. Could you try your linter on the output that the weight bot pushed please?

I will see what I can do, we use this as a template for our benchmarks and I am not sure if we used that part of the code.

**Comment by simonsso** [comment]
I have updated the lines which had an impact to the linter in our project.

File: frame/lottery/src/weights.rs [hunk 3]
```diff
@@ -167,8 +168,8 @@ impl WeightInfo for () {
 	// Storage: Lottery TicketsCount (r:1 w:1)
 	// Storage: Lottery Tickets (r:1 w:0)
 	fn on_initialize_end() -> Weight {
-		// Minimum execution time: 62_720 nanoseconds.
-		Weight::from_ref_time(63_545_000 as u64)
+		// Minimum execution time: 61_712 nanoseconds.
+		Weight::from_ref_time(62_144_000 as u64)
```
Comment (simonsso) [inline frame/lottery/src/weights.rs:172]: This as u64 is still present here, should be gone in 769c2e4f3634f71a2f13914ea3c50f34eda40754

**Comment by ggwpez** [comment]
/cmd queue -c bench-bot $ pallet dev pallet_lottery

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
bot rebase

File: frame/lottery/src/weights.rs (hunk 3 above)
Comment (ggwpez) [reply frame/lottery/src/weights.rs]: Should be good now, thanks!

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**